### ISSUE TITLE
in-vitro model not working on newer scikit-learn

### ISSUE THREAD

**Issue by miles-weatherseed** [pdxgx/pepsickle#3]
in-vitro-2 works fine for me but at the model loading stage of in-vitro I get the following error: 

`/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/base.py:348: InconsistentVersionWarning: Trying to unpickle estimator DummyClassifier from version 0.23.2 when using version 1.3.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:
https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations
  warnings.warn(
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.11/bin/pepsickle", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/pepsickle/__init__.py", line 78, in main
    cleavage_model = initialize_digestion_gb_model(human_only=
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/pepsickle/model_functions.py", line 208, in initialize_digestion_gb_model
    model = joblib.load(_model_path)
            ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/joblib/numpy_pickle.py", line 658, in load
    obj = _unpickle(fobj, filename, mmap_mode)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/joblib/numpy_pickle.py", line 577, in _unpickle
    obj = unpickler.load()
          ^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/pickle.py", line 1213, in load
    dispatch[key[0]](self)
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/joblib/numpy_pickle.py", line 415, in load_build
    self.stack.append(array_wrapper.read(self))
                      ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/joblib/numpy_pickle.py", line 252, in read
    array = self.read_array(unpickler)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/joblib/numpy_pickle.py", line 152, in read_array
    array = pickle.load(unpickler.file_handle)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "sklearn/tree/_tree.pyx", line 728, in sklearn.tree._tree.Tree.__setstate__
  File "sklearn/tree/_tree.pyx", line 1434, in sklearn.tree._tree._check_node_ndarray
ValueError: node array from the pickle has an incompatible dtype:
- expected: {'names': ['left_child', 'right_child', 'feature', 'threshold', 'impurity', 'n_node_samples', 'weighted_n_node_samples', 'missing_go_to_left'], 'formats': ['<i8', '<i8', '<i8', '<f8', '<f8', '<i8', '<f8', 'u1'], 'offsets': [0, 8, 16, 24, 32, 40, 48, 56], 'itemsize': 64}
- got     : [('left_child', '<i8'), ('right_child', '<i8'), ('feature', '<i8'), ('threshold', '<f8'), ('impurity', '<f8'), ('n_node_samples', '<i8'), ('weighted_n_node_samples', '<f8')]`

Have tried a few tricks but can't seem to fix it myself. Would be fantastic if you get the chance to have a look as I was finding the model very helpful!

**Comment by weederb23** [comment]
hi @miles-weatherseed, 

sorry for the slow reply. I believe I may know what the issue is here... let me check a few things and see if I can resolve the problem. 

In a recent update we standardized the versioning for the default models, however the in-vitro model specifically is trained using a slightly different framework and wasn't updated automatically in the same way the default models were.

A temporary fix should be to use a virtual environment (conda, etc.) with the older version of sklearn installed (0.23.2) which was originally used for model training and generation, however I will see if we can fix this more permanently with a package/model update.

**Comment by weederb23** [comment]
@miles-weatherseed this should be fixed with version 0.2.2 as long as you're using the accompanying conda environment. The environment version number has changed to 0.2.2 for consistency, however if you were using the previous yaml version the packages have all stayed the same and have not changed.

Please let me know if this fixes your issue when you have a chance!

**Comment by miles-weatherseed** [comment]
@weederb23 Hi ben, thanks for taking a look. I was quite keen to avoid using the conda environment (just a personal preference - I lean more towards virtualenv). I was able to get the model working by pickling it and loading it with scikit-learn 1.2.0. For some reason, when using the joblib I got an error: AttributeError: 'GradientBoostingClassifier' object has no attribute '_loss'. Did you mean: 'loss'? This disappeared when using pickle.
